Feature/Complete functionality to delete documents

psbhatbvbcs opened the pull request:
Fixes #4 : Unable to delete documents

Hey @jacoblee93 and @Nutlope ! Amazing project.

- This feature successfully handles deletion of documents completely. 
- I have made sure to add functionality such that the document is also deleted from BYTESCALE cloud where the pdf was being stored. 
- It also deletes the record from the Prism-Postgres database.
- The document list also dynamically updates after deletion.
- Everything is pretty much dynamic for scaling

NOTE:
- For this feature to work I had to add an additional .env variable from BYTESCALE. As the current Bytescale key was a public api key it had limited permissions and couldn't delete files. 
- Add the below variable to the .env file:
NEXT_SECRET_BYTESCALE_API_KEY=

This key can be found in: Bytescale Dashboard > Security > API keys > Secret.

I plan to make regular contributions on the project if I am welcome!
Thank you! And hope you find my feature helpful. :smile:

jacoblee93 commented on app/api/chat/route.ts:
```diff
@@ -170,6 +170,8 @@ export async function POST(req: NextRequest) {
       ),
     ).toString('base64');
 
+    // add a chat thing to prisma
```
Remove

psbhatbvbcs replied:
Yup. Removed in the new commit

jacoblee93 commented on app/api/deleteDocument/route.ts:
```diff
@@ -0,0 +1,96 @@
+
+interface DeleteFileParams {
+  accountId: string;
+  apiKey: string;
```
Does this need to be a parameter?

psbhatbvbcs replied:
I followed the docs of bytescale to implement functionality so I just added this to resolve typescript errors. The Api key is taken from the env variables and then passed to this. Can be removed but i though of keeping the same syntax as docs.

jacoblee93 commented on app/dashboard/dashboard-client.tsx:
```diff
@@ -70,6 +87,35 @@ export default function DashboardClient({ docsList }: { docsList: any }) {
+        console.log(data.error);
+      } else {
+        console.log('Document deleted successfully');
+        updateDocsList(docsList.filter((doc) => doc.id !== id));
```
Probably unnecessary to wrap this? Just call `setDocsList` directly?

psbhatbvbcs replied:
Yeah right. I have done all the requested changes

jacoblee93 commented:
I think logic looks good! @Nutlope would be better to weigh in on the frontend UX and design.

psbhatbvbcs commented:
@jacoblee93 I have done all the requested changes! Thanks for bringing them into light

Eric013 commented:
Hello,

@jacoblee93 To complete the deletion of a document, shouldn't the vectors created in pinecone also be deleted?
`await index.namespace('foo-namespace').deleteAll();`
See: https://sdk.pinecone.io/typescript/#md:delete-all-records-in-a-namespace

jacoblee93 commented:
You are correct, thanks for catching!

psbhatbvbcs commented:
Hey @Eric013 !! I though of that too initially, but I had assumed it just created new vectors from 0 when i added new documents. But after checking you were right. Added that functionality too to delete that specific document vectors with its namespace.
